server/src/main/java/org/opensearch/tasks/TaskManager.java

```diff
@@ -448,6 +467,41 @@ public void waitForTaskCompletion(Task task, long untilInNanos) {
         throw new OpenSearchTimeoutException("Timed out waiting for completion of [{}]", task);
     }
 
+    /**
+     * Adds Task Id in the ThreadContext.
+     * <p>
+     * Stashes the existing ThreadContext and preserves all the existing ThreadContext's data in the new ThreadContext
+     * as well.
+     *
+     * @param task for which Task Id needs to be added in ThreadContext.
+     * @return StoredContext reference to restore the ThreadContext from which we created a new one.
+     * Caller can call context.restore() to get the existing ThreadContext back.
+     */
+    public ThreadContext.StoredContext addTaskIdInThreadContext(@Nullable Task task) {
+        if (task == null) {
+            return () -> {};
+        }
+
+        ThreadContext threadContext = threadPool.getThreadContext();
```

Comment:
Not really sure if ``TaskManager`` should be the real owner for updating the thread context. Can it not be done directly by the tracking service instead once the task is created?

Reply:
Yes, Moving to tracking service